src/storage/mvcc/txn.rs

```diff
@@ -160,17 +160,18 @@ impl<'a> MvccTxn<'a> {
                 (lock.lock_type, lock.short_value.take())
             }
             _ => {
-                return match try!(self.reader.get_txn_commit_ts(key, self.start_ts)) {
-                    // Committed by concurrent transaction.
-                    Some(_) => Ok(()),
-                    // Rollbacked by concurrent transaction.
+                return match try!(self.reader.get_txn_commit_info(key, self.start_ts)) {
+                    Some((_, WriteType::Rollback)) |
```

Comment:
We check commit_info only when lock is not exist. So when lock is not exist, and data had been marked with committed(is rollback or not exist), return `TxnLockNotFound`.